app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,215 @@
+module EmsRefresh
+  module Parsers
+    class Azure < ManageIQ::Providers::CloudManager::RefreshParser
+      include Vmdb::Logging
+
+      VALID_LOCATION = /\w+/
+
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, options = nil)
+        @ems        = ems
+        config      = ems.connect
+        @vmm        = ::Azure::Armrest::VirtualMachineService.new(config)
+        @asm        = ::Azure::Armrest::AvailabilitySetService.new(config)
+        @options    = options || {}
+        @data       = {}
+        @data_index = {}
+      end
+
+      def ems_inv_to_hashes
+        log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS : [#{@ems.name}] id: [#{@ems.id}]"
```

Comment:
The `MIQ(#{self.class.name}.#{__method__})` is actually redundant here ... the new `_log` logger will insert that for you.